main.rb

```diff
@@ -0,0 +1,114 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    return to_enum unless block_given?
+
+    entry = is_a?(Range) ? to_a : self
+    i = 0
+    while i < size
+      yield(entry[i])
+      i += 1
+    end
+    entry
+  end
+
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    entry = is_a?(Range) ? to_a : self
+    i = 0
+    while i < size
+      yield(entry[i], i)
+      i += 1
+    end
+    entry
+  end
+
+  def my_select
+    return to_enum unless block_given?
+
+    arr_new = []
+    my_each { |elem| arr_new << elem if yield(elem) }
+    arr_new
+  end
+
+  def my_all?(arg = nil)
+    if block_given?
+      my_each { |elem| return false unless yield(elem) }
+      return true
+    end
+    if arg.nil?
+      my_each { |elem| return false unless elem }
+      return true
+    end
+    my_each { |elem| return false unless check_validity(elem, arg) }
+    true
+  end
+
+  def my_any?(arg = nil, &proc)
+    if block_given?
+      my_each { |elem| return true if proc.nil? ? proc.call(elem) : yield(elem) }
+    else
+      my_each { |elem| return true if arg.nil? ? elem : check_validity(elem, arg) }
+    end
+    false
+  end
```

Comment:
**These too**
- [ ] when a class is passed as an argument returns true if at least one of the collection is a member of such class
- [ ] when a Regex is passed as an argument returns false if none of the collection matches the Regex
- [ ] when a pattern other than Regex or a Class is given returns false if none of the collection matches the pattern